app/models/manageiq/providers/vmware/infra_manager/refresher.rb

```diff
@@ -118,6 +118,10 @@ def get_vc_data(ems, accessors = VC_ACCESSORS_HASH, mor_filters = {})
           @vc_data[:storage_profile_datastore] = @vi.pbmQueryMatchingHub(storage_profile_ids)
         end
 
+        storage_profile_ids = @vc_data[:storage_profile].collect { |uid, _profile| uid }
+        @vc_data[:storage_profile_datastore] = @vi.pbmQueryMatchingHub(storage_profile_ids)
+        @vc_data[:storage_profile_entity]    = @vi.pbmQueryAssociatedEntity(storage_profile_ids)
```

Comment:
I think this should be moved up into the block above, see [here](https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/vmware/infra_manager/refresher.rb#L117)